### PR TITLE
Document Pkcs12SafeContents

### PR DESCRIPTION
**Pull request by bartonjs**
cc: @carlossanlop

### CODE REVIEW

File: xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml [hunk 1]
```diff
@@ -33,7 +33,7 @@
       </AssemblyInfo>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
+        <summary>Initializes a new instance of the <see cref="T:System.Security.Cryptography.Pkcs12.Pkcs12SafeContents"/> class.</summary>
```
Comment (bartonjs) [inline xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml]:
```suggestion
        <summary>Initializes a new instance of the <see cref="T:System.Security.Cryptography.Pkcs.Pkcs12SafeContents"/> class.</summary>
```

File: xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml [hunk 2]
```diff
@@ -57,10 +57,14 @@
...
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="certificate">The certificate to add.</param>
+        <summary>Adds a certificate to the SafeContents via a new <see cref="T:System.Security.Cryptography.Pkcs.Pkcs12CertBag"/> and returns the newly created bag instance.</summary>
+        <returns>The bag instance which was added to the SafeContents.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentNullException">The <paramref name="certificate"/> parameter is <see langword="null"/>.</exception>
+        <exception cref="T:System.InvalidOperationException">This instance is read-only.</exception>
+        <exception cref="T:System.Security.Cryptography.CryptographicException">The <paramref name="certificate"/> parameter is in an invalid state.</exception>
+        <altmember cref="P:System.Security.Cryptography.Pkcs12.Pkcs12SafeContents.IsReadOnly" />
```
Comment (bartonjs) [inline xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml]:
```suggestion
        <altmember cref="P:System.Security.Cryptography.Pkcs.Pkcs12SafeContents.IsReadOnly" />
```

File: xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml [hunk 3]
```diff
@@ -83,10 +87,16 @@
...
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="key">The asymmetric private key to add.</param>
+        <summary>Adds an asymmetric private key to the SafeContents via a new <see cref="T:System.Security.Cryptography.Pkcs.Pkcs12KeyBag"/> and returns the newly created bag instance.</summary>
+        <returns>The bag instance which was added to the SafeContents.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentNullException">The <paramref name="key"/> parameter is <see langword="null"/>.</exception>
+        <exception cref="T:System.InvalidOperationException">This instance is read-only.</exception>
+        <exception cref="T:System.Security.Cryptography.CryptographicException">The key export failed.</exception>
+        <altmember cref="P:System.Security.Cryptography.Pkcs12.Pkcs12SafeContents.IsReadOnly" />
```
Comment (bartonjs) [inline xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml]:
```suggestion
        <altmember cref="P:System.Security.Cryptography.Pkcs.Pkcs12SafeContents.IsReadOnly" />
```

File: xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml [hunk 4]
```diff
@@ -83,10 +87,16 @@
...
+        <altmember cref="Overload:System.Security.Cryptography.Pkcs12.Pkcs12SafeContents.AddShroudedKey" />
```
Comment (bartonjs) [inline xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml]:
```suggestion
        <altmember cref="Overload:System.Security.Cryptography.Pkcs.Pkcs12SafeContents.AddShroudedKey" />
```

File: xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml [hunk 5]
```diff
@@ -189,12 +223,18 @@
...
+        <param name="key">The asymmetric private key to add.</param>
+        <param name="passwordBytes">The bytes to use as a password when encrypting the key material.</param>
+        <param name="pbeParameters">The password-based encryption (PBE) parameters to use when encrypting the key material.</param>
+        <summary>Adds an encrypted asymmetric private key to the SafeContents via a new <see cref="T:System.Security.Cryptography.Pkcs.Pkcs12ShroudedKeyBag"/> from a byte-based password in an array and returns the newly created bag instance.</summary>
+        <returns>The bag instance which was added to the SafeContents.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentNullException">The <paramref name="key"/> parameter is <see langword="null"/>.</exception>
+        <exception cref="T:System.InvalidOperationException">This instance is read-only.</exception>
+        <exception cref="T:System.Security.Cryptography.CryptographicException">The key export failed.</exception>
+        <altmember cref="P:System.Security.Cryptography.Pkcs12.Pkcs12SafeContents.IsReadOnly" />
```
Comment (bartonjs) [inline xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml]:
```suggestion
        <altmember cref="P:System.Security.Cryptography.Pkcs.Pkcs12SafeContents.IsReadOnly" />
```

File: xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml [hunk 6]
```diff
@@ -189,12 +223,18 @@
...
+        <altmember cref="Overload:System.Security.Cryptography.Pkcs12.Pkcs12SafeContents.AddShroudedKey" />
```
Comment (bartonjs) [inline xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml]:
```suggestion
        <altmember cref="Overload:System.Security.Cryptography.Pkcs.Pkcs12SafeContents.AddShroudedKey" />
```

File: xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml [hunk 7]
```diff
@@ -218,12 +258,18 @@
...
+        <param name="key">The asymmetric private key to add.</param>
+        <param name="passwordBytes">The bytes to use as a password when encrypting the key material.</param>
+        <param name="pbeParameters">The password-based encryption (PBE) parameters to use when encrypting the key material.</param>
+        <summary>Adds an encrypted asymmetric private key to the SafeContents via a new <see cref="T:System.Security.Cryptography.Pkcs.Pkcs12ShroudedKeyBag"/> from a byte-based password in a span and returns the newly created bag instance.</summary>
+        <returns>The bag instance which was added to the SafeContents.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentNullException">The <paramref name="key"/> parameter is <see langword="null"/>.</exception>
+        <exception cref="T:System.InvalidOperationException">This instance is read-only.</exception>
+        <exception cref="T:System.Security.Cryptography.CryptographicException">The key export failed.</exception>
+        <altmember cref="P:System.Security.Cryptography.Pkcs12.Pkcs12SafeContents.IsReadOnly" />
```
Comment (bartonjs) [inline xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml]:
```suggestion
        <altmember cref="P:System.Security.Cryptography.Pkcs.Pkcs12SafeContents.IsReadOnly" />
```

File: xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml [hunk 8]
```diff
@@ -218,12 +258,18 @@
...
+        <altmember cref="Overload:System.Security.Cryptography.Pkcs12.Pkcs12SafeContents.AddShroudedKey" />
```
Comment (bartonjs) [inline xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml]:
```suggestion
        <altmember cref="Overload:System.Security.Cryptography.Pkcs.Pkcs12SafeContents.AddShroudedKey" />
```

File: xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml [hunk 9]
```diff
@@ -247,12 +293,18 @@
...
+        <param name="key">The asymmetric private key to add.</param>
+        <param name="password">The password to use when encrypting the key material.</param>
+        <param name="pbeParameters">The password-based encryption (PBE) parameters to use when encrypting the key material.</param>
+        <summary>Adds an encrypted asymmetric private key to the SafeContents via a new <see cref="T:System.Security.Cryptography.Pkcs.Pkcs12ShroudedKeyBag"/> from a char-based password in a span and returns the newly created bag instance.</summary>
+        <returns>The bag instance which was added to the SafeContents.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentNullException">The <paramref name="key"/> parameter is <see langword="null"/>.</exception>
+        <exception cref="T:System.InvalidOperationException">This instance is read-only.</exception>
+        <exception cref="T:System.Security.Cryptography.CryptographicException">The key export failed.</exception>
+        <altmember cref="P:System.Security.Cryptography.Pkcs12.Pkcs12SafeContents.IsReadOnly" />
```
Comment (bartonjs) [inline xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml]:
```suggestion
        <altmember cref="P:System.Security.Cryptography.Pkcs.Pkcs12SafeContents.IsReadOnly" />
```

File: xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml [hunk 10]
```diff
@@ -247,12 +293,18 @@
...
+        <altmember cref="Overload:System.Security.Cryptography.Pkcs12.Pkcs12SafeContents.AddShroudedKey" />
```
Comment (bartonjs) [inline xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml]:
```suggestion
        <altmember cref="Overload:System.Security.Cryptography.Pkcs.Pkcs12SafeContents.AddShroudedKey" />
```

File: xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml [hunk 11]
```diff
@@ -276,12 +328,18 @@
...
+        <param name="key">The asymmetric private key to add.</param>
+        <param name="password">The password to use when encrypting the key material.</param>
+        <param name="pbeParameters">The password-based encryption (PBE) parameters to use when encrypting the key material.</param>
+        <summary>Adds an encrypted asymmetric private key to the SafeContents via a new <see cref="T:System.Security.Cryptography.Pkcs.Pkcs12ShroudedKeyBag"/> from a char-based password in a string and returns the newly created bag instance.</summary>
+        <returns>The bag instance which was added to the SafeContents.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentNullException">The <paramref name="key"/> parameter is <see langword="null"/>.</exception>
+        <exception cref="T:System.InvalidOperationException">This instance is read-only.</exception>
+        <exception cref="T:System.Security.Cryptography.CryptographicException">The key export failed.</exception>
+        <altmember cref="P:System.Security.Cryptography.Pkcs12.Pkcs12SafeContents.IsReadOnly" />
```
Comment (bartonjs) [inline xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml]:
```suggestion
        <altmember cref="P:System.Security.Cryptography.Pkcs.Pkcs12SafeContents.IsReadOnly" />
```

File: xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml [hunk 12]
```diff
@@ -276,12 +328,18 @@
...
+        <altmember cref="Overload:System.Security.Cryptography.Pkcs12.Pkcs12SafeContents.AddShroudedKey" />
```
Comment (bartonjs) [inline xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml]:
```suggestion
        <altmember cref="Overload:System.Security.Cryptography.Pkcs.Pkcs12SafeContents.AddShroudedKey" />
```

File: xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml [hunk 13]
```diff
@@ -163,9 +194,12 @@
       <Docs>
         <param name="secretType">To be added.</param>
```
Comment (rpetrusha) [inline xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml]: Can you document these two parameters?

File: xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml [hunk 14]
```diff
@@ -163,9 +194,12 @@
...
-        <returns>To be added.</returns>
+        <summary>Adds an ASN.1 BER-encoded value with a specified type identifier to the SafeContents via a new <see cref="T:System.Security.Cryptography.Pkcs.Pkcs12SecretBag"/> and returns the newly created bag instance.</summary>
+        <returns>The bag instance which was added to the SafeContents.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ArgumentNullException">The <paramref name="secretType"/> parameter is <see langword="null"/>.</exception>
+        <exception cref="T:System.InvalidOperationException">This instance is read-only.</exception>
+        <exception cref="T:System.Security.Cryptography.CryptographicException">The <paramref name="secretValue"/> parameter does not represent a single ASN.1 BER-encoded value.</exception>
       </Docs>
     </Member>
     <Member MemberName="AddShroudedKey">
```
Comment (rpetrusha) [inline xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml]: You've done a really good job of disambiguating the method overload summaries.

File: xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml [hunk 15]
```diff
@@ -247,12 +293,18 @@
...
       <Docs>
-        <param name="key">To be added.</param>
-        <param name="password">To be added.</param>
-        <param name="pbeParameters">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="key">The asymmetric private key to add.</param>
+        <param name="password">The password to use when encrypting the key material.</param>
+        <param name="pbeParameters">The password-based encryption (PBE) parameters to use when encrypting the key material.</param>
+        <summary>Adds an encrypted asymmetric private key to the SafeContents via a new <see cref="T:System.Security.Cryptography.Pkcs.Pkcs12ShroudedKeyBag"/> from a char-based password in a span and returns the newly created bag instance.</summary>
```
Comment (rpetrusha) [inline xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml]:
```suggestion
        <summary>Adds an encrypted asymmetric private key to the SafeContents via a new <see cref="T:System.Security.Cryptography.Pkcs.Pkcs12ShroudedKeyBag"/> from a character-based password in a span and returns the newly created bag instance.</summary>
```

File: xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml [hunk 16]
```diff
@@ -276,12 +328,18 @@
...
       <Docs>
-        <param name="key">To be added.</param>
-        <param name="password">To be added.</param>
-        <param name="pbeParameters">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="key">The asymmetric private key to add.</param>
+        <param name="password">The password to use when encrypting the key material.</param>
+        <param name="pbeParameters">The password-based encryption (PBE) parameters to use when encrypting the key material.</param>
+        <summary>Adds an encrypted asymmetric private key to the SafeContents via a new <see cref="T:System.Security.Cryptography.Pkcs.Pkcs12ShroudedKeyBag"/> from a char-based password in a string and returns the newly created bag instance.</summary>
```
Comment (rpetrusha) [inline xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml]:
```suggestion
        <summary>Adds an encrypted asymmetric private key to the SafeContents via a new <see cref="T:System.Security.Cryptography.Pkcs.Pkcs12ShroudedKeyBag"/> from a character-based password in a string and returns the newly created bag instance.</summary>
```

File: xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml [hunk 17]
```diff
@@ -376,9 +462,22 @@
         <Parameter Name="password" Type="System.ReadOnlySpan&lt;System.Char&gt;" />
       </Parameters>
       <Docs>
-        <param name="password">To be added.</param>
-        <summary>To be added.</summary>
-        <remarks>To be added.</remarks>
+        <param name="password">The password to use for decrypting the encrypted contents.</param>
+        <summary>Decrypts the contents of this SafeContents value using a char-based password from a span.</summary>
```
Comment (rpetrusha) [inline xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml]:
```suggestion
        <summary>Decrypts the contents of this SafeContents value using a character-based password from a span.</summary>
```

File: xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml [hunk 18]
```diff
@@ -401,9 +500,22 @@
         <Parameter Name="password" Type="System.String" />
       </Parameters>
       <Docs>
-        <param name="password">To be added.</param>
-        <summary>To be added.</summary>
-        <remarks>To be added.</remarks>
+        <param name="password">The password to use for decrypting the encrypted contents.</param>
+        <summary>Decrypts the contents of this SafeContents value using a char-based password from a string.</summary>
```
Comment (rpetrusha) [inline xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml]:
```suggestion
        <summary>Decrypts the contents of this SafeContents value using a character-based password from a string.</summary>
```

File: xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml [hunk 19]
```diff
@@ -424,9 +536,11 @@
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <summary>Gets an enumerable representation of the SafeBag values contained within the SafeContents.</summary>
+        <returns>An enumerable representation of the SafeBag values contained within the SafeContents.</returns>
         <remarks>To be added.</remarks>
+        <exception cref="T:System.InvalidOperatinException">The contents are encrypted.</exception>
```
Comment (rpetrusha) [inline xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml]:
```suggestion
        <exception cref="T:System.InvalidOperationException">The contents are encrypted.</exception>
```

File: xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml (hunk 13 above)
Comment (bartonjs) [inline xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml]:
```suggestion
         <param name="secretType">The Object Identifier (OID) which identifies the data type of the secret value.</param>
```

File: xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml [hunk 20]
```diff
@@ -163,9 +194,12 @@
       <Docs>
         <param name="secretType">To be added.</param>
         <param name="secretValue">To be added.</param>
```
Comment (bartonjs) [inline xml/System.Security.Cryptography.Pkcs/Pkcs12SafeContents.xml]:
```suggestion
         <param name="secretValue">The BER-encoded value representing the secret to add.</param>
```

**Comment by rpetrusha** [comment]
The build was successful, so I'll merge this now, @bartonjs.

**Pull request merged**